base/string.jl

```diff
@@ -968,8 +968,8 @@ byte_string_classify(s::ByteString) = byte_string_classify(s.data)
     # 1: valid ASCII
     # 2: valid UTF-8
 
-is_valid_ascii(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
-is_valid_utf8(s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) != 0
+isvalid(::Type{ASCIIString}, s::Union(Array{UInt8,1},ByteString)) = byte_string_classify(s) == 1
```

Comment:
It strikes me that we could check for valid ASCII much faster than calling `byte_string_classify`.  e.g.
```
isvalid(::Type{ASCIIString}, s::ByteString) = isvalid(ASCIIString, s.data)
function isvalid(::Type{ASCIIString}, s::Array{UInt8,1})
    for c in s
        c >= 128 && return false
    end
    return true
end